packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/DuplicateObjectEventHandler.java

```diff
@@ -0,0 +1,234 @@
+        this.objectService = Objects.requireNonNull(objectService);
+        this.messageService = Objects.requireNonNull(messageService);
+
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER).tag(Monitoring.NAME, this.getClass().getSimpleName()).register(meterRegistry);
+    }
+
+    @Override
+    public boolean canHandle(IEditingContext editingContext, IInput input) {
+        return input instanceof DuplicateObjectInput;
+    }
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IInput input) {
+        this.counter.increment();
+
+        List<Message> messages = List.of(new Message(this.messageService.invalidInput(input.getClass().getSimpleName(), DuplicateObjectInput.class.getSimpleName()), MessageLevel.ERROR));
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, editingContext.getId(), input);
+        IPayload payload = new ErrorPayload(input.id(), messages);
+
+        if (input instanceof DuplicateObjectInput duplicateObjectInput) {
+            IStatus duplicationResult = this.duplicateObject(editingContext, duplicateObjectInput.objectId(), duplicateObjectInput.containerId(),
+                    duplicateObjectInput.containmentFeatureName(), duplicateObjectInput.duplicateContent(), duplicateObjectInput.copyOutgoingReferences(),
+                    duplicateObjectInput.updateIncomingReferences());
+
+            if (duplicationResult instanceof Success success) {
+                payload = new DuplicateObjectSuccessPayload(input.id(), success.getParameters().get(NEW_OBJECT), success.getMessages());
+                changeDescription = new ChangeDescription(success.getChangeKind(), editingContext.getId(), input);
+            } else if (duplicationResult instanceof Failure failure) {
+                payload = new ErrorPayload(input.id(), failure.getMessages());
+            }
+        }
+
+        payloadSink.tryEmitValue(payload);
+        changeDescriptionSink.tryEmitNext(changeDescription);
+    }
+
+    private IStatus duplicateObject(IEditingContext editingContext, String objectToDuplicateId, String containerId, String containmentFeature, boolean duplicateContent,
+            boolean copyOutgoingReferences, boolean updateIncomingReferences) {
+
+        IStatus result;
+        Optional<EObject> objectToDuplicateOpt = this.objectService.getObject(editingContext, objectToDuplicateId)
+                .filter(EObject.class::isInstance)
+                .map(EObject.class::cast);
+        Optional<EObject> containerEObjectOpt = this.objectService.getObject(editingContext, containerId)
+                .filter(EObject.class::isInstance)
+                .map(EObject.class::cast);
+
+        if (objectToDuplicateOpt.isEmpty()) {
+            result = new Failure(MessageFormat.format("The object with id \"{0}\" does not exist", objectToDuplicateId));
```

Comment:
Use the message service instead (I think we already have a message in case something does not exist)